static/app/views/releases/list/index.tsx

```diff
@@ -180,6 +184,51 @@ class ReleasesList extends DeprecatedAsyncView<Props, State> {
     return projects?.find(p => p.id === `${selectedProjectId}`);
   }
 
+  getAllSelectedProjects(): Project[] {
+    const {selection, projects} = this.props;
+    return projects.filter(
+      project =>
+        selection.projects.indexOf(parseInt(project.id, 10)) > -1 ||
+        selection.projects.indexOf(-1) > -1
+    );
+  }
+
+  getAllEnvironments(): string[] {
+    const {selection, projects} = this.props;
+    const selectedProjects = selection.projects;
+    const {user} = ConfigStore.getState();
+    const allEnvSet = new Set(projects.flatMap(project => project.environments));
+    // NOTE: mostly taken from environmentSelector.tsx
+    const unSortedEnvs = new Set(
+      projects.flatMap(project => {
+        const projectId = parseInt(project.id, 10);
```

Comment:
if you keep the project ids as strings you wouldn't need to parse them everywhere

Reply:
The issue here is that we're also populating this from the api request which returns the body as a JSON object with strings i believe.

This is mainly to be consistent

Reply:
Why convert them to numbers in the first place? We're not doing math on them right?

Reply:
I think because the PageFitlers `selection` types them as numbers?

Reply:
I learned [this](https://stackoverflow.com/questions/23556873/different-answer-in-javascript-and-python-with-large-number-division-or-mod-ope) in the hard way that backend ints are not the same as frontend ones 😅 best practice is to keep ids as string so things don't break as company scales.

Reply:
Yup - this is mainly relevant when we're operating on large numbers
This shouldn't be an issue with our ids though as i don't think our ids are nearly that large.

Since `PageFilters` is typing these as numbers, i can't quite modify that response without doing a whole refactor.
Instead i'll cast the filter to a string for the checks